changes/Hop_sequence (modular).py

```diff
@@ -0,0 +1,133 @@
+    
+
+# we craete a class hop_func() that have attributes as limit_range, max_stars, limiting_brightness, click
+# hence for every click this class have functions that return a hop which can be stored in a hop_sequence 
+# to generate a star plot.
+
+# this value is used to avoid returning of hops even when there is no nearby star (e.g. editor clicking
+# somewhere near the borders).
+
+# max_stars is the limit put to remove unnessecory faint stars from our star list
+
+# limiting_brightness is the minimum brightness below which we consider a star to be faint and is used to 
+# remove stars unnecessary faint stars if the no of stars go up by max_stars
+
+class hop_func():
+  
+  def __init__(self, limit_range, max_stars, limiting_brightness, x, y):
+    self.limit_range = limit_range
+    self.max_stars = max_stars
+    self.limiting_brightness = limiting_brightness
+    self.click = Point(x, y)
+
+  # the function list_star() returns a list of stars that lie 
+  # within the sqaure formed with the click as center
+  
+  def list_stars(self, data):
+    stars_catalogue = pd.DataFrame(columns = ['star','ra','dec','brightness'])
+
+    l_x_limit = int(self.click.x) - self.limit_range
+    r_x_limit = int(self.click.x) + self.limit_range
+    l_y_limit = int(self.click.y) - self.limit_range
+    r_y_limit = int(self.click.y) + self.limit_range
+  
+    for i in range(len(data)):    
+      if (data['ra'][i] > l_x_limit and data['ra'][i] < r_x_limit) and (data['dec'][i] > l_y_limit and data['dec'][i] < r_y_limit):
+        stars_catalogue = stars_catalogue.append({'star': data['star'][i], 'ra': data['ra'][i], 'dec': data['dec'][i], 'brightness': data['brightness'][i]}, ignore_index=True)
+
+    if len(stars_catalogue) >= self.max_stars:
+       stars_catalogue = stars_catalogue[stars_catalogue['brightness'] > self.limiting_brightness]
+
+    return stars_catalogue
+
+  # the function min_distance() will return the value of the nearest 
+  # star to the clicks from the selected stars.  
+  
+  def min_distance(self, star_catalogue):
+    star_name = None
+
+    stars_point = []
+    # star_point is list of Point objects with ra as 
+    # the x and dec as the y values of the star.
+    for i in range(len(star_catalogue)):
+      x = star_catalogue['ra'][i]
+      y = star_catalogue['dec'][i]
+      p = Point(x,y)
+      stars_point.append(p)
+  
+    try:
+      min_val = self.click.distance(stars_point[0])
+      min_star = stars_point[0]
+  
+      count = 0
+      # to keep a track of stars which are at a same value from our click
+  
+      for i in range(len(stars_point)):
+        if min_val >= self.click.distance(stars_point[i]):
+          min_val = self.click.distance(stars_point[i])
+          min_star = stars_point[i]
+          star_name = star_catalogue['star'][i]
+          count += 1
+    
+      assert(count == 1)
+
+      return min_star, star_name
+  
+    except AssertionError:
```

Comment:
@vedshenoy should we choose the closest star when we find two or more points found in the vicinity?